src/thermal_throttle.c

```diff
@@ -0,0 +1,100 @@
+/**
+ * collectd - src/thermal_throttle.c
+ * Copyright (C) 2017      notbaab
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; only version 2 of the License is applicable.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ * Authors:
+ *   notbaab <notbaab at gmail.com>
+ **/
+
+#include "common.h"
+#include "plugin.h"
+#include "collectd.h"
+
+static int num_cpu = 0;
+
+static int thermal_throttle_init() {
```

Comment:
I'd prefer to do the `if (access() == 0) { break; }` check in the main read function, rather than counting the (expected) number of CPUs at startup. The number of active CPUs may vary over the runtime of the daemon.

Reply:
Yes, that's what I'm recommending.

If that's the only thing you do in the init function, then you can remove it entirely – it's not mandatory.

Best regards,
—octo